File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ans = ListNode()
        dummy = ans
        rem = 0
        while l1 or l2:
            if l1 and l2:
                node = l1.val + l2.val + rem
                l1 = l1.next
                l2 = l2.next
            elif l1:
                node = l1.val + rem
                l1 = l1.next
            else:
                node = l2.val + rem
                l2 = l2.next
            dummy.next = ListNode(node % 10)
            dummy = dummy.next
            rem = int(node / 10)
        if rem > 0: dummy.next = ListNode(rem)
        return ans.next
    
        

File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    let ans = new ListNode(0);
    let head = ans
    let rem = 0;
    while (l1 || l2) {
        let node = rem;
        if (l1) {
            node += l1.val
            l1 = l1.next
        }
        if (l2) {
            node += l2.val
            l2 = l2.next
        } 
        ans.next = new ListNode(node % 10)
        rem = Math.floor(node / 10)
        ans = ans.next
    }
    if (rem > 0) ans.next = new ListNode(rem)
    return head.next
};